internal/provider/worker.go

```diff
@@ -0,0 +1,280 @@
+package provider
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/hashicorp/boundary/api"
+	"github.com/hashicorp/boundary/api/workers"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+const (
+	scope                              = "scope"
+	scopeId                            = "global"
+	createdTime                        = "created_time"
+	version                            = "version"
+	address                            = "address"
+	canonicalTags                      = "canonical_tags"
+	configTags                         = "config_tags"
+	workerGeneratedAuthToken           = "worker_generated_auth_token"
+	controllerGeneratedActivationToken = "controller_generated_activation_token"
+	apiTags                            = "api_tags"
+	releaseVersion                     = "release_version"
+	authorizedActions                  = "authorized_actions"
+)
+
+func resourceWorker() *schema.Resource {
+	return &schema.Resource{
+		Description: "The resource allows you to create a self-managed worker object.",
+
+		CreateContext: resourceWorkerCreate,
+		ReadContext:   resourceWorkerRead,
+		UpdateContext: resourceWorkerUpdate,
+		DeleteContext: resourceWorkerDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			IDKey: {
+				Description: "The ID of the worker.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			ScopeIdKey: {
+				Description: "The scope for the worker.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			NameKey: {
+				Description: "The name for the worker.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			DescriptionKey: {
+				Description: "The description for the worker.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			address: {
+				Description: "The accessible address of the self managed worker.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			canonicalTags: {
+				Description: "The aggregated view of worker tags and API tags.",
+				Type:        schema.TypeMap,
+				Elem: &schema.Schema{
+					Type: schema.TypeList,
+					Elem: &schema.Schema{
+						Type: schema.TypeString,
+					},
+				},
+				Computed: true,
+			},
+			configTags: {
+<<<<<<< HEAD:internal/provider/worker.go
+				Description: "Tags as configured in the worker's HCL file.",
+=======
+				Description: "",
+>>>>>>> ef6f54e859b702c5a853f3f4e90737190ddf3dcc:internal/provider/resource_self_managed_worker.go
```

Comment:
This looks like a merge conflict that was left in 😅

Reply:
Oops 😂

Reply:
I’ll fix that shortly, but before we merge I’m having some issues with resolving the tag schema. It seems that the map elements should be a list, but that’s not supported in the schema. I’m not sure about the right approach to resolve that. At the moment if someone applies tags to a worker and we refresh, Terraform gets unhappy and nukes the resource from state, but doesn’t delete the resource itself. 
If you have any pointers for me I’d be very appreciative!

Reply:
I’m happy enough to leave tags out so folks can get on with using this (if that makes sense), and then resolve that in a fast follow next week.

Reply:
I agree, we can leave it out.